app/controllers/APIController.scala

```diff
@@ -1,30 +1,132 @@
 package controllers
 
 import services._
-import io.circe.generic.codec.DerivedAsObjectCodec.deriveCodec
 import io.circe.syntax.EncoderOps
+import io.circe._
+
 import javax.inject._
 import play.api._
 import play.api.mvc._
+import DAO.BakeryDB
+import io.circe
+import models.{ProductFromJson, ServiceStatus}
+
+import java.util.UUID
 
 @Singleton
 class APIController @Inject() (
-    val environment: Environment,
-    val controllerComponents: ControllerComponents
+    environment: Environment,
+    val controllerComponents: ControllerComponents,
+    bakeryDB: BakeryDB
 ) extends BaseController {
 
   def statusToJson: String = {
-    val status = models.Status(
+    val status = ServiceStatus(
       "pass-bakery",
       environment.mode.toString,
       APIService.getDateTime
     )
     status.asJson.spaces2
   }
 
+  def parseFromJson(request: Request[AnyContent]): Option[ProductFromJson] = {
+    val parseResult: Option[ProductFromJson] = {
+      request.body.asText match {
+        case None =>
+          None
+        case Some(rawText) => {
+          circe.parser.parse(rawText) match {
+            case Left(failure) =>
+              None
+            case Right(json) =>
+              parser
+                .decode[ProductFromJson](json.toString()) match {
+                case Left(error) =>
+                  None
+                case Right(product) => Some(product)
+              }
+          }
+        }
+      }
+    }
+    parseResult
+  }
+
   def serviceStatus(): Action[AnyContent] = Action {
     implicit request: Request[AnyContent] =>
       Ok(statusToJson)
   }
 
+  def postProduct(): Action[AnyContent] = Action {
+    implicit request: Request[AnyContent] =>
+      parseFromJson(request) match {
+        case None =>
+          BadRequest("Post failed")
+        case Some(product) =>
+          product.name match {
+            case None => BadRequest("Post failed on name")
+            case Some(name) =>
+              product.quantity match {
+                case None => BadRequest("Post failed on quantity")
+                case Some(quantity) =>
+                  product.price match {
+                    case None => BadRequest("Post failed on price")
```

Comment:
If you miss one you fail the entire request. Can you condense this into a three value tuple match?
```scala
(product.name, product.quantity, product.price) match {
    case (Some(name), Some(quantity), Some(price)) =>
    //etc
```